primitives/state-machine/src/basic.rs

```diff
@@ -118,27 +119,31 @@ impl Externalities for BasicExternalities {
 		self.top.get(key).cloned()
 	}
 
-	fn storage_hash(&self, key: &[u8]) -> Option<H256> {
-		self.storage(key).map(|v| Blake2Hasher::hash(&v))
+	fn storage_hash(&self, key: &[u8]) -> Option<Vec<u8>> {
+		self.storage(key).map(|v| Blake2Hasher::hash(&v).encode())
```

Comment:
should we put in the doc that it uses `[u8; 32]` as a hash ?

Reply:
`BasicExternalities` need to be adapted when we have someone that uses a different hash.